src/hotspot/share/prims/jvmtiEnv.cpp

```diff
@@ -3427,14 +3428,23 @@ JvmtiEnv::GetSystemProperty(const char* property, char** value_ptr) {
 // value - NULL is a valid value, must be checked
 jvmtiError
 JvmtiEnv::SetSystemProperty(const char* property, const char* value_ptr) {
-  jvmtiError err =JVMTI_ERROR_NOT_AVAILABLE;
+  NULL_CHECK(property, JVMTI_ERROR_NULL_POINTER);
```

Comment:
You don't need this. The null check happens in the wrapper code generated from the xml file. See  build/<config>/hotspot/variant-server/gensrc/jvmtifiles/jvmtiEnterTrace.cpp

Reply:
I agree with David here that this change is not needed.
This is a relevant fragment from the jvmtiEnter.cpp:
```
static jvmtiError JNICALL
jvmti_SetSystemProperty(jvmtiEnv* env,
            const char* property,
            const char* value_ptr) {
  . . .
  if (Threads::number_of_threads() != 0) {
    . . .
    if (property == NULL) {
      return JVMTI_ERROR_NULL_POINTER;
    }
    err = jvmti_env->SetSystemProperty(property, value_ptr);
  } else {
    if (property == NULL) {
      return JVMTI_ERROR_NULL_POINTER;
    }
    err = jvmti_env->SetSystemProperty(property, value_ptr);
  }
  return err;

```